src/call_frame.rs

```diff
@@ -1,22 +1,16 @@
 use std::num::Saturating;
-use u256::{H160, U256};
+use u256::U256;
 use zkevm_opcode_defs::ethereum_types::Address;
 
 use crate::{state::Stack, store::InMemory};
 
 #[derive(Debug, Clone)]
 pub struct CallFrame {
-    pub heap_id: u32,
-    pub aux_heap_id: u32,
-    pub calldata_heap_id: u32,
-    // Code memory is word addressable even though instructions are 64 bit wide.
-    pub code_page: Vec<U256>,
     pub pc: u64,
     /// Transient storage should be used for temporary storage within a transaction and then discarded.
     pub transient_storage: Box<InMemory>,
```

Comment:
That makes sense, good catch!

Reply:
Reverted because of following cite from primer:

> All near calls inside the contract are sharing the same memory space (heap, stack), and do not roll back the changes to this memory if they fail. They do, however, roll back the changes to storage, transient storage and events.